Tools/scripts/deepfreeze.py

```diff
@@ -5,18 +5,38 @@
 import contextlib
 import os
 import re
+import sys
 import time
 import types
+import unicodedata
 from typing import Dict, FrozenSet, Tuple, TextIO
 
 import umarshal
 
 verbose = False
 
+# See Objects/unicodectype.c:_PyUnicode_IsPrintable()
+NON_PRINTABLE = {"Cc", "Cf", "Cs", "Co", "Cn", "Zl", "Zp", "Zs"}
+
+
+def isprintable(b: bytes) -> bool:
+    """isascii() and isprintable() for Python 3.6
+
+    return b.isascii() and b.decode("ascii").isprintable()
+    """
+    try:
+        s = b.decode("ascii")
+    except UnicodeDecodeError:
+        return False
+    for c in s:
+        if unicodedata.category(c) in NON_PRINTABLE:
+            return False
+    return True
```

Comment:
What I really wanted is this:

```suggestion

def isprintable(b: bytes) -> bool:
    return all(0x20 <= c < 0x7f for c in b)
```

There's only one call site and it just uses this to decide whether to use hex escapes for string literals in the generated output. Unicode character classes feel like overkill. :-)

Reply:
$deity bless ASCII-only land! It's so much easier. :)